packages/client-config/API.md

```diff
@@ -206,7 +207,10 @@ export type CustomClientConfig = {
 export const DEFAULT_CLIENT_CONFIG_VERSION: ClientConfigVersion;
 
 // @public
-export const generateClientConfig: <T extends "1" | "0">(credentialProvider: AwsCredentialIdentityProvider, backendIdentifier: DeployedBackendIdentifier, version: T) => Promise<ClientConfigVersionTemplateType<T>>;
+export const generateClientConfig: <T extends "0" | "1">(options: GenerateClientConfigOptions, backendIdentifier: DeployedBackendIdentifier, version: T) => Promise<ClientConfigVersionTemplateType<T>>;

```

Comment:
Should `options` be last parameter and optional?
If not provided should these just fall back to default clients/node credential chain?

Reply:
It's not optional since `credentialProvider` is part of the options, and that's the fallback.

Reply:
`credentialProvider` could be made optional and fall back to default node credentail chain.

Reply:
If it is safe/desirable to fall back to `fromNodeProviderChain` why do we pass `credentialsProvider` as an option at all?

I believe we talked about this a few months ago, but would it make more sense at that point to allow `region` to be passed in alongside `credentialsProvider` instead?

Reply:
> If it is safe/desirable to fall back to fromNodeProviderChain 

It should be safe. @edwardfoyle @Amplifiyer please correct me if I'm wrong. 

> why do we pass credentialsProvider as an option at all?

I'm not 100% sure at this point, but I think the parameter was added to make it explicit that caller should be aware what creds they're using for this. I think we should retain option to provide it even if we make it optional.

> I believe we talked about this a few months ago, but would it make more sense at that point to allow region to be passed in alongside credentialsProvider instead?

I think it's good idea. SDK accepts both. See example here: https://github.com/aws-amplify/amplify-backend/blob/1e93535be2d46112a23d0f2cd8736de1238a3795/packages/integration-tests/src/e2e_tooling_client_config.ts#L17-L18